src/main/java/us/kbase/narrativemethodstore/db/mongo/MongoUtils.java

```diff
@@ -58,81 +58,28 @@ public static <T> List<T> getProjection(
-     * The object must be deserializable by an {@link ObjectMapper} configured so private
-     * fields are visible.
-     * @param dbo the MongoDB object to transform.
+
+    /** Map a MongoDB {@link Document} to a class.
+     * @param doc the MongoDB document to transform.
      * @param clazz the class to which the object will be transformed.
      * @return the transformed object.
      */
-    public static <T> T toObject(final DBObject dbo, final Class<T> clazz) {
-        return dbo == null ? null : MAPPER.convertValue(toMap(dbo), clazz);
-    }
-    
-    /** Map a MongoDB {@link BSONObject} to a standard map.
-     * This method expects that all maps and lists in the objects are implemented as
-     * {@link BSONObject}s or derived classes, not standard maps, lists, or other classes.
-     * @param dbo the MongoDB object to transform to a standard map.
-     * @return the transformed object, or null if the argument was null.
-     */
-    public static Map<String, Object> toMap(final BSONObject dbo) {
-        @SuppressWarnings("unchecked")
-        final Map<String, Object> ret = (Map<String, Object>) cleanObject(dbo);
-        return ret;
-    }
-    
-    // this assumes there aren't BSONObjects embedded in standard objects, which should
-    // be the case for stuff returned from mongo
-    
-    // Unimplemented error for dbo.toMap()
-    // dbo is read only
-    // can't call ObjectMapper.convertValue() on dbo since it has a 'size' field outside of
-    // the internal map
-    // and just weird shit happens when you do anyway
-    private static Object cleanObject(final Object dbo) {
-        // sometimes it's lazy, sometimes it's basic. Not sure when or why.
-        if (dbo instanceof LazyBSONList) {
-            final List<Object> ret = new LinkedList<>();
-            // don't stream, sometimes has issues with nulls
-            for (final Object obj: (LazyBSONList) dbo) {
-                ret.add(cleanObject(obj));
-            }
-            return ret;
-        } else if (dbo instanceof BasicBSONList) {
-            final List<Object> ret = new LinkedList<>();
-            // don't stream, sometimes has issues with nulls
-            for (final Object obj: (BasicBSONList) dbo) {
-                ret.add(cleanObject(obj));
-            }
-        } else if (dbo instanceof BSONObject) {
-            // can't stream because streams don't like null values at HashMap.merge()
-            final BSONObject m = (BSONObject) dbo;
-            final Map<String, Object> ret = new HashMap<>();
-            for (final String k: m.keySet()) {
-                if (!k.equals("_id")) {
-                    final Object v = m.get(k);
-                    ret.put(k, cleanObject(v));
-                }
-            }
-            return ret;
-        }
-        return dbo;
+    public static <T> T toObject(final Document doc, final Class<T> clazz) {
+        return doc == null ? null : MAPPER.convertValue(doc, clazz);
```

Comment:
Hmm, I'm not completely sure this is going to work for all scenarios. If the test coverage was better I'd be less concerned

Reply:
toObject only has 1 usage. Which scenario do you think might not work?

Reply:
The Jackson mapper doesn't know how to map all objects correctly, so if there's something in doc that isn't compatible with Jackson errors or data corruption might occur

Reply:
How do we address the concern about Jackson's mapping capabilities and move forward effectively? Perhaps we should add a try-catch clause inside the toObject method.

Reply:
covered by `getRepoLastVersion` and `getRepoLastVersionFail`

Reply:
I'm not sure what you mean by that. Is the entire call tree covered?

Reply:
https://app.codecov.io/gh/kbase/narrative_method_store/pull/99/blob/src/main/java/us/kbase/narrativemethodstore/db/mongo/MongoDynamicRepoDB.java#L331

Reply:
My understanding of a call tree is that it shows which methods call other methods. I am not sure exactly what you are asking.

Reply:
Right. So what I'm asking is whether all the methods that ultimately, either directly or indirectly, call toObject are covered, starting from the application entrypoint. If that's the case then we can be reasonably confident (although not sure) there aren't any surprises

Reply:
<img width="1026" alt="Screenshot 2024-10-28 at 11 31 42 AM" src="https://github.com/user-attachments/assets/a53e9f83-0596-44f1-9c3e-bf38ce2145ff">

Reply:
And those methods are all completely covered?

Reply:
We resolved this offline; the upshot is that they're covered enough

Reply:
:+1: